build/flaws.js

```diff
@@ -55,10 +109,101 @@ function injectFlaws(doc, $) {
         }
       }
     });
-    if (options.flawLevels.get("bad_bcd_queries") === FLAW_LEVELS.ERROR) {
+    if (options.flawLevels.get("broken_links") === FLAW_LEVELS.ERROR) {
       throw new Error(`bad_bcd_queries flaws: ${doc.flaws.bad_bcd_queries}`);
     }
   }
 }
 
-module.exports = { injectFlaws };
+function fixFixableFlaws(doc, options, document) {
+  if (!options.fixFlaws || document.isArchive) return;
+
+  let newRawHTML = document.rawHtml;
+
+  const loud = options.fixFlawsDryRun || options.fixFlawsVerbose;
+
+  // Any 'macros' of type "MacroRedirectedLinkError"...
+  for (const flaw of doc.flaws.macros || []) {
+    if (
+      flaw.name === "MacroRedirectedLinkError" &&
+      (!flaw.filepath || flaw.filepath === document.fileInfo.path)
+    ) {
+      // Sanity check that our understanding of flaws, filepaths, and sources
+      // work as expected.
+      if (!newRawHTML.includes(flaw.macroSource)) {
+        throw new Error(
+          `rawHtml doesn't contain macroSource (${flaw.macroSource})`
+        );
+      }
+      const newMacroSource = flaw.macroSource.replace(
+        flaw.redirectInfo.current,
+        flaw.redirectInfo.suggested
+      );
+      // Remember, in JavaScript only the first occurrence will be replaced.
+      newRawHTML = newRawHTML.replace(flaw.macroSource, newMacroSource);
+      if (loud) {
+        console.log(
+          chalk.grey(
+            `Fixed macro ${chalk.white.bold(
+              flaw.macroSource
+            )} to ${chalk.white.bold(newMacroSource)}`
+          )
+        );
+      }
+      flaw.fixed = true;
+    }
+  }
+
+  // Any 'broken_links' with a suggestion...
+  for (const flaw of doc.flaws.broken_links || []) {
+    if (flaw.suggestion) {
```

Comment:
code style nit: I like early returns (or `continue` in this case) to reduce nesting and (imo) aid readability. So in this case it'd mean changing that line to `if (!flaw.suggestion) { continue; }`

Reply:
Sure. A bit less indentation will taste good here.